rustyms/src/glycan/render/shape.rs

```diff
@@ -0,0 +1,531 @@
+                outer_modifications,
+            ]
+            .join("")
+        };
+        match &self.base_sugar {
+            BaseSugar::Pentose(isomer) => (
+                Shape::Star,
+                match isomer {
+                    None | Some(PentoseIsomer::Xylulose) => Colour::Background,
+                    Some(PentoseIsomer::Arabinose) => Colour::Green,
+                    Some(PentoseIsomer::Lyxose) => Colour::Yellow,
+                    Some(PentoseIsomer::Xylose) => Colour::Orange,
+                    Some(PentoseIsomer::Ribose) => Colour::Pink,
+                },
+                inner_modifications,
+                outer_mods(
+                    nacetyl,
+                    acid,
+                    amino,
+                    deoxy,
+                    acetyl,
+                    glycolyl,
+                    nglycolyl,
+                    o_carboxy_ethyl,
+                ),
+            ),
+            BaseSugar::Hexose(isomer) => {
+                if o_carboxy_ethyl > 0 && nacetyl > 0 {
+                    (
+                        Shape::Hexagon,
+                        Colour::Purple,
+                        inner_modifications,
+                        outer_mods(
+                            nacetyl - 1,
+                            acid,
+                            amino,
+                            deoxy,
+                            acetyl,
+                            glycolyl,
+                            nglycolyl,
+                            o_carboxy_ethyl - 1,
+                        ),
+                    )
+                } else if o_carboxy_ethyl > 0 && nglycolyl > 0 {
+                    (
+                        Shape::Hexagon,
+                        Colour::LightBlue,
+                        inner_modifications,
+                        outer_mods(
+                            nacetyl,
+                            acid,
+                            amino,
+                            deoxy,
+                            acetyl,
+                            glycolyl,
+                            nglycolyl - 1,
+                            o_carboxy_ethyl - 1,
+                        ),
+                    )
+                } else if o_carboxy_ethyl > 0 && amino > 0 {
+                    (
+                        Shape::Hexagon,
+                        Colour::Brown,
+                        inner_modifications,
+                        outer_mods(
+                            nacetyl,
+                            acid,
+                            amino - 1,
+                            deoxy,
+                            acetyl,
+                            glycolyl,
+                            nglycolyl,
+                            o_carboxy_ethyl - 1,
+                        ),
+                    )
+                } else if deoxy > 1 {
+                    let c = match isomer {
+                        Some(HexoseIsomer::Glucose) => Colour::Blue,
```

Comment:
These color look-ups could be moved to a function or a macro, they recur a lot.

Reply:
I would very much like to, but the thing is that every time they are different. Unless for example `GalNAc` can be drawn as a yellow divided triangle, which is currently defined as a white deivided triangle. I see now that some could maybe be reused I will try and do that.